doc/source/getting_started/notebooks/Exploring Vector Spaces.ipynb

```diff
@@ -0,0 +1,883 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "The angle between [1.0, 1.0, 0.0] and [1.0, 0.0, 0.0] is 45.00000000000001 degrees.\n"
+     ]
+    }
+   ],
+   "source": [
+    "RAD2DEG = 180 / np.pi  # conversion factor to degrees\n",
+    "x = space.element([1, 1, 0])\n",
+    "y = space.element([1, 0, 0])\n",
+    "\n",
+    "cos_angle_rad = x.inner(y) / np.sqrt(x.inner(x) * y.inner(y))\n",
+    "angle_rad = np.arccos(cos_angle_rad)\n",
+    "angle_deg = angle_rad * RAD2DEG\n",
+    "print('The angle between {} and {} is {} degrees.'.format(x, y, angle_deg))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Norms"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": true
+   },
+   "source": [
+    "A norm takes a vector in $\\mathbb{R}^n$ and maps it to a postive number:\n",
+    "\n",
+    "$$\n",
+    "    \\|\\cdot\\| : \\mathbb{R}^n \\to [0, \\infty)\n",
+    "$$\n",
+    "\n",
+    "It is *positively one-homogeneous*, *positive definite* and satisfies the *triangle inequality*, see <a href=\"https://en.wikipedia.org/wiki/Norm_(mathematics)\">here</a> for details. The standard norm on $\\mathbb{R}^n$ is the root of the sum of squares of all components:\n",
+    "\n",
+    "$$\n",
+    "    \\|x\\|_2 = \\sqrt{\\sum_{k=1}^n x_k^2}\n",
+    "$$\n",
+    "\n",
+    "This is just the same as $\\sqrt{\\langle x, x\\rangle}$ -- the norm is *induced* by the inner product.\n",
+    "\n",
+    "Again, this function is available as a space method `space.norm(vec)` or as method on the element itself, `vec.norm()`:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "x = space.element([2, 3, 6])  # has norm sqrt(2*2 + 3*3 + 6*6) = 7"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "7.0"
+      ]
+     },
+     "execution_count": 10,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "space.norm(x)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "7.0"
+      ]
+     },
+     "execution_count": 11,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "x.norm()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "There is a variety of norms on $\\mathbb{R}^n$ that are not induced by an inner product. They are parametrized by a real number $p \\in [1, \\infty]$ and are defined as\n",
```

Comment:
There is way more norms than this (in fact, I've actually been looking at a reference on what norms exist.) You can imagine norms like

|| (x1, x2, x3) || = max(|x1|, ||(x2, x3)||_2)

etc